### PR TITLE
refactor: clean non webpack public APIs

### PR DESCRIPTION
**Pull request by h-a-n-a**
## Summary

Cleaned non webpack public APIs to avoid misalignment.
Refactor the current filename to constructor style.

Related: https://github.com/web-infra-dev/rspack/issues/3726

## Test Plan

Passing the original tests.

### CODE REVIEW

File: packages/rspack/src/Compilation.ts [hunk 1]
```diff
@@ -103,11 +103,22 @@ export class Compilation {
 	logging: Map<string, LogEntry[]>;
 	name?: string;
 	childrenCounters: Record<string, number> = {};
+	/**
+	 * Note: This is not a webpack public API, maybe removed in future.
```
Comment (Boshen) [inline packages/rspack/src/Compilation.ts:107]: nitpick: comment on why is this needed, and the requirement for removal. Having "maybe" does not communicate well.

Comment (h-a-n-a) [reply packages/rspack/src/Compilation.ts:107]: Leaving a phrase like "do not use" maybe too aggressive, so I used "maybe" to make the tone softer. What else would you recommend?
